### PR TITLE
remove pk check and fix genericRelation deletion

### PR DESCRIPTION
**Pull request by codehia**

### CODE REVIEW

File: softdelete/models.py [hunk 1]
```diff
@@ -170,7 +164,12 @@ def set_deleted(self, d):
     deleted = property(get_deleted, set_deleted)
 
     def _do_delete(self, changeset, related):
-        rel = related.get_accessor_name()
+        get_rel_name = getattr(related, 'get_accessor_name', getattr(related, 'get_attname', ''))
```
Comment (sandizzle) [inline softdelete/models.py:167]: What is get_attname?

Comment (codehia) [reply softdelete/models.py:167]: for genericRelation fields, this method returns the attribute name of the genericRelation for the model object.

File: softdelete/models.py [hunk 2]
```diff
@@ -237,13 +236,13 @@ def delete(self, *args, **kwargs):
             all_related = [
                 f for f in self._meta.get_fields()
                 if (f.one_to_many or f.one_to_one)
-                   and f.auto_created and not f.concrete
+                   and (f.auto_created or hasattr(f, 'bulk_related_objects')) and not f.concrete
```
[the remark this answers is not in the document]
Comment (codehia) [reply softdelete/models.py:239]: `bulk_related_objects` returns all the related objects with genericRelation to the model. This method is exclusive to GenericRelations,

File: softdelete/models.py [hunk 3]
```diff
@@ -237,13 +236,13 @@ def delete(self, *args, **kwargs):
...
             ]
 
             for x in all_related:
-                if x.on_delete.__name__ not in ['DO_NOTHING', 'SET_NULL']:
+                if hasattr(x, 'bulk_related_objects') or x.on_delete.__name__ not in ['DO_NOTHING', 'SET_NULL']:
                     self._do_delete(cs, x)
-                if x.on_delete.__name__ == 'SET_NULL':
+                if hasattr(x, 'on_delete') and x.on_delete.__name__ == 'SET_NULL':
```
[the remark this answers is not in the document]
Comment (codehia) [reply softdelete/models.py:245]: yes. I will try to separate the conditions for more readability here.

File: softdelete/models.py [hunk 4]
```diff
@@ -237,13 +238,13 @@ def delete(self, *args, **kwargs):
...
-                   and f.auto_created and not f.concrete
+                   and (f.auto_created or hasattr(f, 'bulk_related_objects')) and not f.concrete
             ]
 
             for x in all_related:
-                if x.on_delete.__name__ not in ['DO_NOTHING', 'SET_NULL']:
+                if hasattr(x, 'bulk_related_objects') or (x.on_delete.__name__ not in ['DO_NOTHING', 'SET_NULL']):
                     self._do_delete(cs, x)
-                if x.on_delete.__name__ == 'SET_NULL':
+                elif x.on_delete.__name__ == 'SET_NULL':
```
Comment (sandizzle) [inline softdelete/models.py:247]: Why change if -> elif?

**Review by sandizzle — APPROVED** [review]

**Pull request merged**